### PR TITLE
Create ZipBuilder CI to compile tooling

### PR DESCRIPTION
**Pull request by ryanwilson**
This doesn't test, but does ensure that the zip tooling actually builds.

#no-changelog

### CODE REVIEW

**Comment by ryanwilson** [comment]
From an earlier build, here's an uploaded artifact :)

zip_built

This *should* only build the zip during cron jobs, but will have to verify tomorrow to make sure it still ran as expected.

File: .github/workflows/zip.yml [hunk 1]
```diff
@@ -9,8 +13,29 @@ jobs:
   build:
     runs-on: macOS-latest
     steps:
-    - uses: actions/checkout@v1
+    - uses: actions/checkout@v2
```
Comment (paulb777) [inline .github/workflows/zip.yml]: What does v2 mean versus v1?  I haven't been able to find this in the docs?

File: .github/workflows/zip.yml [hunk 2]
```diff
@@ -9,8 +13,29 @@ jobs:
...
+    - name: Build
+      run: |
+        cd ZipBuilder
+        swift build -v
+
+  package:
+    needs: build
+    runs-on: macOS-latest
+    # Only run the `package` job for the cron job, not PRs.
+    if: github.event_name == 'schedule'
```
Comment (paulb777) [inline .github/workflows/zip.yml]: Nice find!

File: .github/workflows/zip.yml [hunk 3]
```diff
@@ -9,8 +13,29 @@ jobs:
...
     - name: ZipBuildingTest
-      run: ./scripts/build_zip.sh
+      run: |
+         mkdir -p zip_output_dir
+         sh -x scripts/build_zip.sh zip_output_dir
+
+    - uses: actions/upload-artifact@v1
+      with:
+        name: Firebase-actions-dir
+        # Zip the entire output directory since the builder adds subdirectories we don't know the
```
Comment (paulb777) [inline .github/workflows/zip.yml]: Isn't this the directory the zip file is copied into?

File: .github/workflows/zip.yml (hunk 1 above)
Comment (ryanwilson) [reply .github/workflows/zip.yml]: It's now the default for starting a new action - looks like Dec 13, 2019 it was released: https://github.com/actions/checkout/releases/tag/v2.0.0 

We should probably update the rest of the actions, but should do so in a separate PR.

File: .github/workflows/zip.yml (hunk 2 above)
Comment (ryanwilson) [reply .github/workflows/zip.yml]: Thanks! Hopefully it works... got this from the `if` syntax:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/workflow-syntax-for-github-actions#jobsjob_idif

and the "`github` Context" https://help.github.com/en/actions/automating-your-workflow-with-github-actions/contexts-and-expression-syntax-for-github-actions#github-context

hoping that the name matches with the event trigger here:

https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#scheduled-events-schedule

File: .github/workflows/zip.yml (hunk 3 above)
Comment (ryanwilson) [reply .github/workflows/zip.yml]: Not the case due to the Kokoro setup - the output dir is the root directory and artifacts get packaged in this structure:

```
# Zip is named `Firebase-latest-manual.zip` when no RC number is passed in.
<buildDir>/MAJOR_MINOR_PATCH/Firebase-latest-manual.zip
<buildDir>/carthage/MAJOR.MINOR.PATCH/latest-non-rc/<ALL_CARTHAGE_FILES>
```

This is used for the automated Kokoro setup, so unless we know the Firebase version we can't predict the full path of the output. I'm also fairly certain that regex doesn't work yet on the artifacts fetching, so we can't do something like `<buildDir>/*/Firebase-latest-manual.zip` currently.

**Pull request merged**